es6rest-persistence/src/main/java/com/netflix/conductor/dao/es6rest/Elasticsearch6RestModule.java

```diff
@@ -38,19 +38,22 @@ public RestHighLevelClient getClient(Configuration config) throws Exception {
         }
 
         HttpHost[] hosts = Arrays.stream(clusterAddress.split(","))
-                .map(HttpHost::create)
-                .toArray(HttpHost[]::new);
+            .map(HttpHost::create)
+            .toArray(HttpHost[]::new);
 
         int timeout = config.getIntProperty("workflow.elasticsearch.timeout.seconds", 60) * 1000;
-        RestClientBuilder builder = RestClient.builder(hosts).setMaxRetryTimeoutMillis(timeout)
-                .setRequestConfigCallback(new RequestConfigCallback() {
-                    @Override
-                    public RequestConfig.Builder customizeRequestConfig(RequestConfig.Builder requestConfigBuilder) {
-                        return requestConfigBuilder.setConnectionRequestTimeout(0).setSocketTimeout(timeout)
-                                .setConnectTimeout(timeout);
-                    }
-                });
+        return RestClient.builder(hosts).setMaxRetryTimeoutMillis(timeout)
+            .setRequestConfigCallback(new RequestConfigCallback() {
+                @Override
+                public RequestConfig.Builder customizeRequestConfig(RequestConfig.Builder requestConfigBuilder) {
+                    return requestConfigBuilder.setConnectionRequestTimeout(0).setSocketTimeout(timeout)
+                        .setConnectTimeout(timeout);
+                }
+            });
+    }
 
+    @Provides
+    public RestHighLevelClient getClient(Configuration config, RestClientBuilder builder) {
```

Comment:
The main change is here is that `RestHighLevelClient` instance will be created for the each DAO module (Queue, Execution, Metadata) instead of having a single across the JVM.